app/main.py

```diff
@@ -1,2 +1,58 @@
 class Dictionary:
-    pass
+    def __init__(self):
+        self.length = 0
+        self.hash_table = [[] for _ in range(8)]
+        self.capacity = len(self.hash_table)
+
+    def __setitem__(self, key, value):
+        self._resize()
+        index = hash(key) % self.capacity
+        for _ in self.hash_table:
```

Comment:
Why use `for` loop here
When you don't know the number of iterations `while` is a more common solution